virtual_ecosystem/models/abiotic/energy_balance.py

```diff
@@ -538,67 +532,42 @@ def calculate_leaf_and_air_temperature(
     )
     new_air_temperature = a_A + b_A * delta_canopy_temperature
     new_canopy_temperature = (
-        (data["air_temperature"][1 : true_canopy_layers_n + 1]).to_numpy()
+        (data["air_temperature"][layer_structure.index_filled_canopy]).to_numpy()
         + delta_canopy_temperature
     )
 
     # Interpolate temperature below canopy
-    # This could potentially be done without explicit below canopy layers
-    target_heights = np.vstack(
-        [
-            data["layer_heights"]
-            .where(data["layer_heights"]["layer_roles"] == "subcanopy")
-            .dropna(dim="layers"),
-            data["layer_heights"]
-            .where(data["layer_heights"]["layer_roles"] == "surface")
-            .dropna(dim="layers"),
-        ]
-    )
+
+    # TODO - This only uses the index of the _last_ filled layer, which works with the
+    #        current test where the canopy layers are consistent across cells, but will
+    #        break with uneven canopy layers.
+
```

Comment:
This is why I used the function to find the lowest values for each grid cell at some point, that should make it work with varying number of canopy layers. But does not need to be addressed here

Reply:
There is probably a better way to do that now that we no longer have the subcanopy layer

Reply:
Yeah - the LayerStructure will be able to provide specific coordinates to extract the target values using `lowest_filled_layer`, but may be able to do it with numpy array commands:

```python
In [128]: y = np.arange(25).reshape((5,5))

In [129]: y
Out[129]: 
array([[ 0,  1,  2,  3,  4],
       [ 5,  6,  7,  8,  9],
       [10, 11, 12, 13, 14],
       [15, 16, 17, 18, 19],
       [20, 21, 22, 23, 24]])

In [130]: y[tuple([np.arange(5),np.arange(4, -1, -1)])]
Out[130]: array([ 4,  8, 12, 16, 20])
```